### PR TITLE
Add Difficulty Rating to Questions

### LINKED ISSUE #606: [Frontend] Implement Question difficulty rating UI
## Description

We want users to be able to rate question difficulty in the UI. 

**Blocker**: There is no API for this as of yet. So this issue is blocked on the backend task.

### PR DESCRIPTION
**Pull request by NazireAta**
## 📋 Proposed Changes

  Added the Difficulty Rating Poll and the Difficulty Bar to the Questions, getting the input from the users to determine community voting for the difficulty level of a question.

## Related Issue
Closes #606

## 🧪 Included Tests
- [ ] The user votes fo the difficulty rating for a question

### CODE REVIEW

File: frontend/src/routes/question.test.tsx [hunk 1]
```diff
@@ -175,4 +200,44 @@ describe("QuestionPage", () => {
...
+                hardCount={2}
+              />
+            }
+          />
+        </Routes>
+      </MemoryRouter>,
+    );
+  
+    // Ensure initial state is rendered
+    expect(screen.getByText("5 Easy votes")).toBeInTheDocument();
```
Comment (mmtftr) [inline frontend/src/routes/question.test.tsx:229]: these texts don't exist in the actual component but are in the test.

File: frontend/src/routes/question.test.tsx [hunk 2]
```diff
@@ -83,6 +93,21 @@ vi.mock("@tanstack/react-query", () => ({
   useQueryClient: vi.fn(),
 }));
 
+vi.mock("@/services/api/programmingForumComponents", () => ({
+  useGetQuestionDetails: vi.fn(() => {}),
```
Comment (mmtftr) [inline frontend/src/routes/question.test.tsx:97]: This mock is overriding the one above it and that's why you're getting the missing mock errors. Merge the contents of this mock with the one above to resolve the issue.

**Review by mmtftr — CHANGES_REQUESTED** [review]
Thanks a lot, I've added comments describing why the tests fail

File: frontend/src/routes/question.test.tsx (hunk 2 above)
Comment (NazireAta) [reply frontend/src/routes/question.test.tsx]: phew! that was harder than i thought it could be

**Review by mmtftr — APPROVED** [review]
Looks good to me, thanks for the work on tests!

**Pull request merged**